File: client/src/dto/MatchContextDto.ts
import { EMPTY_STRING } from "../env";
import { CellDto } from "./CellDto";
import { PlayerResourceBundleDto, undefinedPlayerResourceBundleDto } from "./PlayerInfoBundleDto";

export interface MatchContextDto {
    id: string;
    roomId: string;
    boardArray: CellDto[][];
    currentTurn: number;
    playerInfoBundle: PlayerResourceBundleDto;
}

export const undefinedMatch: MatchContextDto = {
    id: EMPTY_STRING,
    roomId: EMPTY_STRING,
    boardArray: [],
    currentTurn: 0,
    playerInfoBundle: undefinedPlayerResourceBundleDto
};

File: server/dto/server_only/match_context_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.cell_dto import CellDto
from dto.player_dto import PlayerDto


@dataclass
class MatchContextDto(BaseDto):
    id: str
    roomId: str
    boardArray: list[list[CellDto]]
    currentTurn: int
    player1: PlayerDto
    player2: PlayerDto
